x/staking/src/types/domain.rs

```diff
@@ -0,0 +1,329 @@
+pub struct DvvTriplet {
+    pub del_addr: AccAddress,
+    pub val_src_addr: ValAddress,
+    pub val_dst_addr: ValAddress,
+}
+impl DvvTriplet {
+    pub fn new(del_addr: AccAddress, val_src_addr: ValAddress, val_dst_addr: ValAddress) -> Self {
+        Self {
+            del_addr,
+            val_src_addr,
+            val_dst_addr,
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct DvPair {
+    pub val_addr: ValAddress,
+    pub del_addr: AccAddress,
+}
+impl DvPair {
+    pub fn new(val_addr: ValAddress, del_addr: AccAddress) -> Self {
+        Self { val_addr, del_addr }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum BondStatus {
+    Unbonded = 0,
+    Unbonding = 1,
+    Bonded = 2,
+}
+
+impl Display for BondStatus {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            BondStatus::Unbonded => write!(f, "Unbonded"),
+            BondStatus::Unbonding => write!(f, "Unbonding"),
+            BondStatus::Bonded => write!(f, "Bonded"),
+        }
+    }
+}
+
+/// Validator defines a validator, together with the total amount of the
+/// Validator's bond shares and their exchange rate to coins. Slashing results in
+/// a decrease in the exchange rate, allowing correct calculation of future
+/// undelegations without iterating over delegators. When coins are delegated to
+/// this validator, the validator is credited with a delegation whose number of
+/// bond shares is based on the amount of coins delegated divided by the current
+/// exchange rate. Voting power can be calculated as total bonded shares
+/// multiplied by exchange rate.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct Validator {
+    /// operator_address defines the address of the validator's operator; bech encoded in JSON.
+    pub operator_address: ValAddress,
+    /// delegator_shares defines total shares issued to a validator's delegators.
+    pub delegator_shares: Decimal256,
+    /// description defines the description terms for the validator.
+    pub description: Description,
+    /// consensus_pubkey is the consensus public key of the validator, as a Protobuf Any.
+    pub consensus_pubkey: PublicKey,
+    /// jailed defined whether the validator has been jailed from bonded status or not.
+    pub jailed: bool,
+    /// tokens define the delegated tokens (incl. self-delegation).
+    pub tokens: Uint256,
+    /// unbonding_height defines, if unbonding, the height at which this validator has begun unbonding.
+    pub unbonding_height: i64,
+    /// unbonding_time defines, if unbonding, the min time for the validator to complete unbonding.
+    pub unbonding_time: Timestamp,
+    /// commission defines the commission parameters.
+    pub commission: Commission,
+    pub min_self_delegation: Uint256,
+    pub status: BondStatus,
+}
+
+impl Validator {
+    pub fn new_with_defaults(
```

Comment:
Perhaps use Default trait?

Reply:
We cannot use the `Default` trait because many of the fields don't implement the trait. And, it is more important, we need to provide validator address that cannot be a default value and its public key. The cosmos sdk method is named `NewValidator` or in inverted order, I named it `new_with_defaults` because it says that we construct a validator with some default values.

Reply:
:+1:  "construct a validator with some default values." justifies the method.